File: Documentation/Example Scripts/download.rb
#!/usr/bin/env ruby

# Example ruby script for ThreadWatcher by Mr. Freeze, 2010
# gets the url from the browser, then asks for a save location and downloads and wathches the thread

require 'rubygems'
require 'appscript'
include Appscript

tWatcher = Appscript.app.by_id("com.freezeco.threadwatcher")

if tWatcher.is_running?
  if  tWatcher.documents.get.size > 0
    tWatcher.documents.first.newTab
  else
    tWatcher.make(:new => :document)
  end
else
  tWatcher.launch
  # make sure ThreadWatcher has finished launching before we continue
  sleep 1
end

tWatcher.documents.first.tabs.last.watchThreadFromBrowser(:saveSheet => true)
tWatcher.activate

File: scripts/download.py
#!/usr/bin/env /usr/bin/python

# Example python script for ThreadWatcher by Mr. Freeze, 2010
# gets the url from safari, then fetches the thread (does not automatically save it)

from appscript import *
import time

tWatcher = app(id='com.freezeco.threadwatcher')
if tWatcher.isrunning:
	if tWatcher.documents.count() > 0:
		tWatcher.documents.first.newTab()
	else:
		tWatcher.make(new=k.document)
else:
	tWatcher.launch()
	# wait till app finishes launching
	time.sleep(1)
	
# get the url of the front most safari window
safari = app('Safari')
theURL = safari.documents.first.URL.get()

# tell threadwatcher to fetch the thread in the last tab
# of the first window
tWatcher.documents.first.tabs.last.fetch(withURL=theURL)
